### ISSUE TITLE
Unable to use themes to style components that are children of NativeBase.List

### ISSUE THREAD

**Issue by chris-divvito** [GeekyAnts/NativeBase#2281]
- node 8.11.1
 - npm 5.6.0
 - react-native 0.55.4
 - react 16
 - native-base 2.6.1

## Expected behaviour
Able to theme components through the normal means (ejected theme)

## Actual behaviour
Unable to style any components descending from a List

## Steps to reproduce
An example layout:

```
class Test extends React.PureComponent {
  render() {
    return <Container>
      <Content>
        <List>
          <ListItem>
            <Text>I am text</Text>
          </ListItem>
        </List>
      <Content>
    </Container>
  }
}
```

An example theme in `Content.js` of an ejected theme:

```
export default (variables = variable) => {
  const contentTheme = {
    "NativeBase.ListItem": {
      backgroundColor: "#F00",
      "NativeBase.Text": { color: "#00F" }
    }
  };

  return contentTheme;
};
```

Neither the colour, nor the background colour are applied to their respective elements.

## Is the bug present in both iOS and Android or in any one of them?
Both

## Any other additional info which would help us debug the issue quicker.
After fighting this all afternoon, I at least have a way around the issue - It appears there is no theme for the List component, at least in the theme ejected from my current version. This causes... something... to go wrong when shoutem compiles the themes down based on the parentPath.

I added an empty theme to the ejected theme to get this working - that is I added the file `List.js` to the theme folder:

```
import variable from "./../variables/platform";

export default (variables = variable) => ({})
```

and in `index.js` added: 

```
import listTheme from "./List";
// ... snip ...
    "NativeBase.List": {
      ...listTheme(variables)
    },
// ... snip ...
```

but this doesn't do the trick on it's own. I didn't dig deep enough to found out where, but somewhere in the depths of NativeBase and/or shoutem something must be checking for empty objects and filtering them out, because in order for this to work I had to change `List.js` to:

```
import variable from "./../variables/platform";

export default (variables = variable) => ({
  minWidth: 1
});
```

Any other property at all works fine, a long as there is at least one property in the resulting object.

**Comment by chris-divvito** [comment]
@SupriyaKalghatgi I did read the guidelines, and I am not sure that the customizer is relevant here? it's unable to reproduce the kind of styling I am speaking about in the issue.

**Comment by SupriyaKalghatgi** [comment]
@chris-divvito Did you find this working?

**Comment by chris-divvito** [comment]
Nope, not without the workaround as stated in the original issue posting

**Comment by suvenduchhatoi** [comment]
@chris-divvito 
If you check NB docs, it has been mentioned that 
"List is deprecated. Use of List for dynamic list generation is discouraged. Use FlatList instead."
Moreover, there is no 'List.js' component within the ejected-theme of native-base because the developers wanted to keep the 'List' more as a standard...So, you should try to customize 'ListItem.js' instead of 'List.js'

**Comment by suvenduchhatoi** [comment]
![screenshot 2018-12-20 at 6 58 18 pm](https://user-images.githubusercontent.com/44160581/50287679-7dda4d00-0489-11e9-86a7-faeb5d4c3c4e.png)

**Comment by suvenduchhatoi** [comment]
![screenshot 2018-12-20 at 6 07 29 pm](https://user-images.githubusercontent.com/44160581/50287694-95b1d100-0489-11e9-9ed9-61aa01585dce.png)
